### PR TITLE
[FIX] product_pricelist_direct_print: fix date context

### PR DESCRIPTION
**Pull request by chienandalu**
This fix a compatibility issue with product_pricelist_supplierinfo if
the wizard date isn't set.

cc @Tecnativa TT32729

ping @carlosdauden @sergio-teruel

### CODE REVIEW

**Review by sergio-teruel — APPROVED** [review]

File: product_pricelist_direct_print/report/product_pricelist_xlsx.py [hunk 1]
```diff
@@ -85,7 +85,9 @@ def _fill_data(self, workbook, sheet, book, pricelist):
                 sheet.write(
                     row,
                     next_col,
-                    product.with_context(pricelist=pricelist.id, date=book.date).price,
+                    product.with_context(
```
Comment (pedrobaeza) [inline product_pricelist_direct_print/report/product_pricelist_xlsx.py]: I think the `None` thing is not obvious, so pointing to the line in core that interprets this value will help later on migrations/refactorings/bugfixes.

Comment (carlosdauden) [reply product_pricelist_direct_print/report/product_pricelist_xlsx.py]: Create a date variable out of bucles and use here:
print_date = book.date or fields.Date.today()

Comment (chienandalu) [reply product_pricelist_direct_print/report/product_pricelist_xlsx.py]: Ok, changes done

**Comment by pedrobaeza** [comment]
/ocabot merge patch

Is this needed in v14?

**Comment by chienandalu** [comment]
> Is this needed in v14?

Well, it should, but I see v14 is missig the whole xlsx stuff :confused:

**Pull request merged**